pkg/apiserver/rest/usecase/workflow.go

```diff
@@ -288,115 +291,251 @@ func (w *workflowUsecaseImpl) DetailWorkflowRecord(ctx context.Context, workflow
+		}
+		if status.Suspend {
+			summaryStatus = model.RevisionStatusSuspend
+		}
+		if status.Terminated {
+			summaryStatus = model.RevisionStatusTerminated
+		}
+
+		record.Status = summaryStatus
+		record.Steps = status.Steps
+		record.Finished = strconv.FormatBool(status.Finished)
+
+		if err := w.ds.Put(ctx, record); err != nil {
+			return err
+		}
+
+		revision.Status = summaryStatus
+		if err := w.ds.Put(ctx, revision); err != nil {
+			return err
+		}
+	}
+
 	return nil
 }
 
-func (w *workflowUsecaseImpl) createWorkflowRecord(ctx context.Context, app *v1beta1.Application, revisionName string) error {
-	status := app.Status.Workflow
+func (w *workflowUsecaseImpl) CreateWorkflowRecord(ctx context.Context, app *v1beta1.Application, revisionName string) error {
+	if app.Annotations == nil {
+		return fmt.Errorf("empty annotations in application")
+	}
+	if app.Annotations[oam.AnnotationWorkflowName] == "" {
+		return fmt.Errorf("failed to get workflow name from application")
+	}
+	if app.Annotations[oam.AnnotationPublishVersion] == "" {
+		return fmt.Errorf("failed to get record version from application")
+	}
 
 	return w.ds.Add(ctx, &model.WorkflowRecord{
 		WorkflowPrimaryKey: app.Annotations[oam.AnnotationWorkflowName],
 		AppPrimaryKey:      app.Name,
-		Name:               strings.TrimPrefix(revisionName, "record-"),
+		RevisionPrimaryKey: revisionName,
+		Name:               app.Annotations[oam.AnnotationPublishVersion],
 		Namespace:          app.Namespace,
-		StartTime:          status.StartTime.Time,
-		Suspend:            status.Suspend,
-		Terminated:         status.Terminated,
-		Steps:              status.Steps,
+		Finished:           "false",
+		StartTime:          time.Now().Time,
+		Status:             model.RevisionStatusInit,
 	})
 }
 
+func (w *workflowUsecaseImpl) ResumeRecord(ctx context.Context, appModel *model.Application, recordName string) error {
+	oamApp, err := w.checkRecordSuspended(ctx, appModel)
+	if err != nil {
+		return err
+	}
+
+	oamApp.Status.Workflow.Suspend = false
+	if err := w.kubeClient.Status().Patch(ctx, oamApp, client.Merge); err != nil {
+		return err
+	}
+	if err := w.syncWorkflowStatus(ctx, oamApp, recordName); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (w *workflowUsecaseImpl) TerminateRecord(ctx context.Context, appModel *model.Application, recordName string) error {
+	oamApp, err := w.checkRecordSuspended(ctx, appModel)
+	if err != nil {
+		return err
+	}
+
+	oamApp.Status.Workflow.Terminated = true
+	if err := w.kubeClient.Status().Patch(ctx, oamApp, client.Merge); err != nil {
+		return err
+	}
+	if err := w.syncWorkflowStatus(ctx, oamApp, recordName); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (w *workflowUsecaseImpl) RollbackRecord(ctx context.Context, appModel *model.Application, recordName, revisionName string) error {
+	oamApp, err := w.checkRecordSuspended(ctx, appModel)
+	if err != nil {
+		return err
+	}
+
+	var record = &model.WorkflowRecord{
```

Comment:
AppPrimaryKey+Name == recordPrimaryKey, please change PrimaryKey function of WorkflowRecord.